pkg/cli/clisqlclient/string_to_duration.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package clisqlclient
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/cockroachdb/errors"
+)
+
+// stringToDuration converts a server-side interval value returned by
+// SHOW LAST QUERY STATISTICS. We use a custom parser here to avoid
+// depending on package `tree`, which makes the SQL shell executable
+// significantly larger.
+func stringToDuration(s string) (time.Duration, error) {
+	// The format is HH:MM:SS[.ffffff]
```

Comment:
this will fail when the HH:MM:SS is in the magnitude of days

Reply:
i'd extract the parseDuration in tree to the duration package in this case.

Reply:
The interval style thing is a problem.

As you've noticed, the CLI code was hardwiring the interval style to POSTGRES. This was nonsense IMHO, because if the user changes the interval style server-side, the CLI code would balk already.

I don't think I want to fix bugs in this PRs that existed already, so I'm not going to do anything about the interval style here. 

I will try to extend the hand-written parser here to support days, however I'd like to point out that we have various hard timeouts inside CockroachDB that prevent queries from running for more than an hour, so that's that. 

The real fix would be to make the SHOW statement returns seconds (as floats) and call it a day. I'll file a separate issue for that.

Reply:
Filed #67618 for  the core issue

Reply:
I'll move forward as discussed here: https://github.com/cockroachdb/cockroach/issues/67618#issuecomment-880113168

Reply:
> As you've noticed, the CLI code was hardwiring the interval style to POSTGRES. This was nonsense IMHO, because if the user changes the interval style server-side, the CLI code would balk already.

It's not a bug because at parse time, there's only a difference with how ParseDInterval works is how negative values work. Timings are never negative. This is captured in that comment.

This change here will become the regression -- ISO8601 and days won't work.